### PR TITLE
Bug fix with check usedKeys

### LINKED ISSUE #21999: Bug fix with check usedKeys
#### Describe the bug

I use webp photo generation and stack next to jpeg file. That is, I always have two files on the server. xxxx.jpg and xxxx.jpg.webp. The second time I added the menu thumbnail to the category, I ran into such a problem that it started creating a file with index 1, although it should have created a file with index 0

#### Steps to Reproduce

1. Add thumbnail menu to any category, for example category id 10
2. Go on server in folder /img/c/10-0_thumb.jpg => /img/c/10-0_thumb.jpg.webp
3. Again add thumbnail menu to category id 10
4. An image will be created with the path: /img/c/10-1_thumb.jpg.webp

If applicable, add screenshots or screen recordings to help explain your problem.

#### Additional information

* PrestaShop version: 1.7.6.8
* PHP version: 7.2

### PR DESCRIPTION
**Pull request by fest73**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | I use webp photo generation and stack next to jpeg file. That is, I always have two files on the server. xxxx.jpg and xxxx.jpg.webp. The second time I added the menu thumbnail to the category, I ran into such a problem that it started creating a file with index 1, although it should have created a file with index 0
| Type?         | bug fix
| Category?     | CO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | Fixes https://github.com/PrestaShop/PrestaShop/issues/21999.
| How to test?  | Here https://github.com/PrestaShop/PrestaShop/issues/21999

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/22000)

### CODE REVIEW

**Review by atomiix — DISMISSED** [review]

File: src/Adapter/Image/Uploader/CategoryMenuThumbnailUploader.php [hunk 1]
```diff
@@ -68,7 +68,7 @@ public function upload($categoryId, UploadedFile $uploadedImage)
         foreach ($files as $file) {
             $matches = [];
 
-            if (preg_match('/^' . $categoryId . '-([0-9])?_thumb.jpg/i', $file, $matches) === 1) {
+            if (preg_match('/^' . $categoryId . '-([0-9])?_thumb.jpg$/i', $file, $matches) === 1) {
```
Comment (Arman-Hosseini) [inline src/Adapter/Image/Uploader/CategoryMenuThumbnailUploader.php]:
```suggestion
            if (preg_match('/^' . $categoryId . '-(\d)?_thumb.jpg$/i', $file, $matches) === 1) {
```

Comment (Arman-Hosseini) [reply src/Adapter/Image/Uploader/CategoryMenuThumbnailUploader.php]: A small change

Comment (Progi1984) [reply src/Adapter/Image/Uploader/CategoryMenuThumbnailUploader.php]: Good idea

**Comment by matthieu-rolland** [comment]
thank you @dariusakafest 👍  could you just add @Arman-Hosseini 's suggestion ? And then your PR is good to go.

**Review by PierreRambaud — APPROVED** [review]

**Review by Progi1984 — APPROVED** [review]

**Review by PululuK — APPROVED** [review]

**Comment by Roomsmaniia** [comment]
Hi,

I can't test this PR, a PM is required to approuve the issue
@PrestaShop/product-team 

Thanks

**Comment by MatShir** [comment]
It's ok for me but it seems that @hibatallahAouadni didn't reproduce the bug. So how are you going to test it @RomainBocheux ?

**Comment by marionf** [comment]
As the issue has been PM :heavy_check_mark: by @MatShir, I remove the waiting for PM label here
I think this PR can go in QA @RomainBocheux  ?

**Comment by marionf** [comment]
The [issue](https://github.com/PrestaShop/PrestaShop/issues/21999) has not been reproduced.
We need to reproduce it before being able to test this PR
@dariusakafest Could you please bring more information in the [issue](https://github.com/PrestaShop/PrestaShop/issues/21999) to help us to reproduce it ?

**Comment by matks** [comment]
Unfortunately the issue https://github.com/PrestaShop/PrestaShop/issues/21999  was closed due to inactivity.

Following, this PR is stale since March.

I close this PR, following our [review guidelines](https://devdocs.prestashop.com/1.7/project/maintainers-guide/reviewing-pull-requests/) "Pull Requests may be closed after 30 days of inactivity". If however you are still interested in it, please ping us :) we can reopen.

Thank you for your contribution.

**Pull request closed without merging**